### ISSUE TITLE
CodePush press install button exit app

### ISSUE THREAD

**Issue by diennguyentien** [Microsoft/react-native-code-push#916]
### Description:
I used code push for my project, when I upgrade to last version in code push i did not finished for update, when I press button install then my application exit. this is my code as below:
```
  componentWillUnmount() {
    AppState.removeEventListener('change', this.handleAppStateChange);
  }

  componentDidMount() {
    AppState.addEventListener('change', this.handleAppStateChange);
    CodePush.sync({
      installMode : CodePush.InstallMode. IMMEDIATE,
    });
  }

  handleAppStateChange(appState) {
    if (appState === 'active') {
      CodePush.sync({installMode: CodePush.InstallMode. IMMEDIATE});
    }
  }
```
Can someone please help me?

### Additional Information

* react-native-code-push version: 3.0.1-beta
* react-native version: 0.46.1
* iOS/Android/Windows version: iOS and Android
* Does this reproduce on a debug build or release build? release build

**Comment by sergey-akhalkov** [comment]
Hi @diennguyentien, thanks for reaching us. As I can see - you are using RN 0.46.1, please note, that we are not supporting this version of RN currently, but we are going to test it and release a new version of react-native-code-push on this week. Could you please try to use RN 0.45.x if possible and verify if the issue is still reproduce?

**Comment by sergey-akhalkov** [comment]
Hi @diennguyentien, we've released [react-native-code-push v4.0.0-beta](https://github.com/Microsoft/react-native-code-push/releases/tag/v4.0.0-beta) that should support RN 0.46.x, please verify and feel free to reopen in case the issue is still reproduce.

**Comment by diennguyentien** [comment]
hi @sergey-akhalkov: thanks you for your support
